code/modules/hydroponics/grown/melon.dm

```diff
@@ -39,6 +40,36 @@
 /obj/item/food/grown/watermelon/make_dryable()
 	return //No drying
 
+/obj/item/food/grown/watermelon/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/kitchen/spoon))
+		return ..()
+
+	var/melon_pulp_count = 1
+	if(seed)
+		melon_pulp_count += round(seed.potency / 25)
+
+	user.balloon_alert(user, "scooped out [melon_pulp_count] pulp")
+	for(var/i in melon_pulp_count)
+		new /obj/item/food/watermelonmush(user.loc)
```

Comment:
I doooo not think this works, unless byond is being fucking weird. you want the `for(var_to_increment in start to end)` syntax